src/transformers/models/owlvit/modeling_tf_owlvit.py

```diff
@@ -0,0 +1,1730 @@
+            outputs = self.owlvit(
+                input_ids=input_ids,
+                pixel_values=pixel_values,
+                attention_mask=attention_mask,
+                output_attentions=output_attentions,
+                output_hidden_states=output_hidden_states,
+            )
+
+            text_model_last_hidden_states = outputs[-2][0]
+            vision_model_last_hidden_states = outputs[-1][0]
+
+        # Embed images and text queries
+        feature_map, query_embeds = self.image_text_embedder(
+            input_ids=input_ids,
+            pixel_values=pixel_values,
+            attention_mask=attention_mask,
+            output_attentions=output_attentions,
+        )
+
+        batch_size, height, width, hidden_dim = feature_map.shape
+        image_feats = tf.reshape(feature_map, (batch_size, height * width, hidden_dim))
+
+        # Reshape from [batch_size * max_text_queries, hidden_dim] -> [batch_size, max_text_queries, hidden_dim]
+        max_text_queries = input_ids.shape[0] // batch_size
+        query_embeds = tf.reshape(query_embeds, (batch_size, max_text_queries, query_embeds.shape[-1]))
+
+        # If first token is 0, then this is a padded query [batch_size, num_queries].
+        input_ids = tf.reshape(input_ids, (batch_size, max_text_queries, input_ids.shape[-1]))
+        query_mask = input_ids[..., 0] > 0
+
+        # Predict object classes [batch_size, num_patches, num_queries+1]
+        (pred_logits, class_embeds) = self.class_predictor(image_feats, query_embeds, query_mask)
+
+        # Predict object boxes
+        pred_boxes = self.box_predictor(image_feats, feature_map)
+
+        if not return_dict:
+            output = (
+                pred_logits,
+                pred_boxes,
+                query_embeds,
+                feature_map,
+                class_embeds,
+                text_model_last_hidden_states,
+                vision_model_last_hidden_states,
+            )
+            output = tuple(x for x in output if x is not None)
+            return output
+
+        return TFOwlViTObjectDetectionOutput(
+            image_embeds=feature_map,
+            text_embeds=query_embeds,
+            pred_boxes=pred_boxes,
+            logits=pred_logits,
+            class_embeds=class_embeds,
+            text_model_last_hidden_states=text_model_last_hidden_states,
+            vision_model_last_hidden_states=vision_model_last_hidden_states,
+        )
+
+    def serving_output(self, output: TFOwlViTObjectDetectionOutput) -> TFOwlViTObjectDetectionOutput:
+        return TFOwlViTObjectDetectionOutput(
+            image_embeds=output.feature_map,
+            text_embeds=output.query_embeds,
+            pred_boxes=output.pred_boxes,
+            logits=output.pred_logits,
+            class_embeds=output.class_embeds,
+            text_model_last_hidden_states=output.text_model_last_hidden_states,
+            vision_model_last_hidden_states=output.vision_model_last_hidden_states,
```

Comment:
Are the hidden_states all the same dimension? If so, can you batch them like in the other `serving_output` methods e.g. `hs = tf.covert_to_tensor(hidden_states) if hidden_states is not None else None`.

If the can't be batched, could you add a comment please?

Reply:
They don't have the same dimensions but I'll add a comment to the code